static/app/views/dashboardsV2/widgetCard/widgetQueries.tsx

```diff
@@ -37,31 +40,79 @@ function transformSeries(stats: EventsStats, seriesName: string): Series {
   };
 }
 
-function transformResult(query: WidgetQuery, result: RawResult): Series[] {
+/**
+ * Multiseries data with a grouping needs to be "flattened" because the aggregate data
+ * are stored under the group names. These names need to be combined with the aggregate
+ * names to show a series.
+ *
+ * e.g. count() and count_unique() grouped by environment
+ * {
+ *    "local": {
+ *      "count()": {...},
+ *      "count_unique()": {...}
+ *    },
+ *    "prod": {
+ *      "count()": {...},
+ *      "count_unique()": {...}
+ *    }
+ * }
+ */
+export function flattenGroupedMultiSeriesData(
+  result: RawResult,
+  queryAlias: string
+): SeriesWithOrdering[] {
+  const seriesWithOrdering: SeriesWithOrdering[] = [];
+  const groupNames = Object.keys(result);
+
+  groupNames.forEach((groupName: string) => {
+    // Each group contains an order key which we should ignore
+    const aggregateNames = Object.keys(omit(result[groupName], 'order'));
+
+    aggregateNames.forEach((aggregate: string) => {
+      const seriesName = `${groupName} : ${aggregate}`;
+      const prefixedName = queryAlias ? `${queryAlias} > ${seriesName}` : seriesName;
```

Comment:
Are we set on this format?
I think we have a bit of inconsistency here between:
- Top 5 Events
- Top 5 Metrics
- This PR
- Widget builder Figma designs (both events and metrics)

Reply:
yes @matejminar is right @narsaynorath . Here some examples:

**Figma:**

![image](https://user-images.githubusercontent.com/29228205/158812243-adde218e-a851-4d94-a6cc-57ee620abb64.png)


**Top 5 Events:**
![image](https://user-images.githubusercontent.com/29228205/158812522-3c7829c2-6c58-447d-9a80-b763ac7395c4.png)

Reply:
we should choose a format and be consistent, but to be honest, I think I like the one you introduced most as it is clear to read

Reply:
Leaving a response here, but I saw you both 👍 'd my comment on slack where there's more context. We're going to have to tackle the tooltip formats and make them more consistent in the future. It's going to be tricky because these are pretty information-dense. Dora is thinking about it!